sql/catalyst/src/main/scala-2.12/org/apache/spark/sql/catalyst/expressions/objects/MapObjectsCodeGenHelper.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst.expressions.objects
+
+import scala.collection.mutable.{Builder, IndexedSeq, WrappedArray}
+
+import org.apache.spark.sql.catalyst.expressions.codegen.CodegenContext
+
+object MapObjectsCodeGenHelper {
```

Comment:
Is it really necessary to have separate copies of this class? Perhaps just looking at the scala version or other properties of the byte code would let you decide on one or the other?

Reply:
I think it is workable if we accept the use `Properties.versionNumberString.startsWith ("2.13")` condition to add `if else branch`branch, I will changed the code to this style first

Reply:
Address 9c36f6c fix this comments, use `Properties.versionNumberString` to decide how to generate code.